server/app/views/api/ApiDocsView.java

```diff
@@ -0,0 +1,270 @@
+import static j2html.TagCreator.pre;
+import static j2html.TagCreator.select;
+import static j2html.TagCreator.span;
+import static j2html.TagCreator.text;
+
+import auth.CiviFormProfile;
+import auth.ProfileUtils;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import j2html.tags.specialized.DivTag;
+import j2html.tags.specialized.OptionTag;
+import j2html.tags.specialized.SelectTag;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.mvc.Http;
+import play.twirl.api.Content;
+import services.CfJsonDocumentContext;
+import services.TranslationNotFoundException;
+import services.export.QuestionJsonSampler;
+import services.program.ProgramDefinition;
+import services.question.LocalizedQuestionOption;
+import services.question.types.MultiOptionQuestionDefinition;
+import services.question.types.QuestionDefinition;
+import views.BaseHtmlLayout;
+import views.BaseHtmlView;
+import views.HtmlBundle;
+import views.admin.AdminLayout;
+import views.admin.AdminLayout.NavPage;
+import views.admin.AdminLayoutFactory;
+
+public class ApiDocsView extends BaseHtmlView {
+  private static final Logger logger = LoggerFactory.getLogger(ApiDocsView.class);
+
+  private final ProfileUtils profileUtils;
+  private final BaseHtmlLayout unauthenticatedlayout;
+  private final AdminLayout authenticatedlayout;
+  private final QuestionJsonSampler.Factory questionJsonSamplerFactory;
+
+  @Inject
+  public ApiDocsView(
+      ProfileUtils profileUtils,
+      BaseHtmlLayout unauthenticatedlayout,
+      AdminLayoutFactory layoutFactory,
+      QuestionJsonSampler.Factory questionJsonSamplerFactory) {
+    this.profileUtils = profileUtils;
+    this.unauthenticatedlayout = unauthenticatedlayout;
+    this.authenticatedlayout = layoutFactory.getLayout(NavPage.API_DOCS);
+    this.questionJsonSamplerFactory = questionJsonSamplerFactory;
+  }
+
+  public Content render(
+      Http.Request request,
+      ProgramDefinition programDefinition,
+      ImmutableSet<String> allProgramSlugs) {
+
+    BaseHtmlLayout layout =
+        isAuthenticatedAdmin(request) ? authenticatedlayout : unauthenticatedlayout;
+
+    HtmlBundle bundle =
+        layout
+            .getBundle(request)
+            .setTitle("API Docs")
+            .addMainContent(contentDiv(programDefinition, allProgramSlugs, request))
+            .addMainStyles("overflow-hidden");
+
+    return layout.render(bundle);
+  }
+
+  private DivTag contentDiv(
+      ProgramDefinition programDefinition,
+      ImmutableSet<String> allProgramSlugs,
+      Http.Request request) {
+
+    SelectTag slugsDropdown =
+        select()
+            .withId("select-slug")
+            .withClasses("border", "border-gray-300", "rounded-lg", "p-2", "ml-2")
+            .attr("onchange", "window.location.href = '/api/docs/v1/' + this.value");
```

Comment:
Unfortunately this is tightly coupled with the `routes.conf`. Since it doesn't change often, it doesn't concern me too much.

It's not possible to get the `routes.ApiDocsController.programForSlug(String).url()` here since we can't dynamically feed in `this.value`; it's a JavaScript variable.